packages/adapters-library/src/adapters/aave-v3/products/rewards/aaveV3RewardsAdapter.ts

```diff
@@ -208,27 +185,41 @@ export class AaveV3RewardsAdapter implements IProtocolAdapter {
       return []
     }
 
-    // Check if user has ever opened a position in AaveV3
     const addressFilter = await this.openPositions(userAddress)
 
-    // Return empty array if user has never opened a AaveV3 position
-    if (!addressFilter?.length) {
+    if (!addressFilter.length) {
       return []
     }
 
-    // Fetch and filter underlying tokens with non-zero rewards in parallel
-    const underlyingTokens = await this.getClaimableUnderlyingTokens(
+    const userRewards = await this.incentivesContract.getAllUserRewards(
+      addressFilter,
       userAddress,
-      protocolToken.underlyingTokens,
-      blockNumber,
+      { blockTag: blockNumber },
+    )
```

Comment:
This function returns all rewards for that user in a single call. It needs the aTokens.